include/dlaf/matrix/panel.h

```diff
@@ -266,8 +281,12 @@ struct Panel<axis, const T, D> {
     const auto panel_coord = dist_matrix_.globalTileFromLocalTile<CoordType>(index.get<CoordType>());
     const GlobalTileIndex panel_index(CoordType, panel_coord);
 
-    const auto size_coord = dist_matrix_.tileSize(panel_index).template get<CoordType>();
-    const auto size_axis = dim_ < 0 ? dist_matrix_.blockSize().template get<axis>() : dim_;
+    const bool is_first_global_tile =
+        dist_matrix_.rankGlobalTile<CoordType>(start_) == dist_matrix_.rankIndex().get(CoordType) &&
+        (start_local_ == index.get(CoordType));
+    const auto size_coord = dist_matrix_.tileSize(panel_index).template get<CoordType>() -
+                            (is_first_global_tile ? start_offset_ : 0);
+    const auto size_axis = dim_ < 0 ? dist_matrix_.size().template get<axis>() : dim_;
```

Comment:
As it can be seen, if `dim_` is not set (=-1), it was returning the blocksize, while now it returns the actual size.

This is a somehow major change in the panel, since it will end up being linked to a sub-panel and not just to a panel of tiles.

This change also required a change access operators (i.e. `read` and `operator()`) in order to deal with this new use-case.

Reply:
What is the rationale of this change?

The maximum size of the panel (and therefore its size if it was not limited by setting it) is a block size. It cannot be larger (as matrix size can be).

Reply:
I understand your concerns. I think it's a good starting point for a discussion.

Let me point out a couple of things, so we can decide how to proceed:
- `Panel` allocates a FULL row/column of tiles;
- When we work at sub-tile level, as in this case, we have to `setWidth`/`setHeight` unconditionally; indeed, we always need just part of the full tile

Somehow, they are faces of the same medal. I would consider reducing the allocation to just what needed, because it is a waste of resources, allowing the `Panel` to work in "sub-tile mode", i.e. it knows that is less than a full tile and returns sub-tiles.

What do you think about this?

Reply:
You can just reduce the block size when creating the panel. Isn't it a better solution?

Reply:
For reference, this topic has been moved to #478, on which this PR is going to be rebased onto.